androidApp/src/main/java/com/example/tasky/android/alarm/data/AlarmScheduler.kt

```diff
@@ -0,0 +1,46 @@
+package com.example.tasky.android.alarm.data
+
+import android.app.AlarmManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import com.example.tasky.agenda.domain.model.AgendaItem
+import com.example.tasky.android.alarm.domain.IAlarmScheduler
+import kotlin.time.DurationUnit
+
+class AlarmScheduler(
+    private val context: Context,
+) : IAlarmScheduler {
+    private val alarmManager = context.getSystemService(AlarmManager::class.java)
+
+    override fun schedule(agendaItem: AgendaItem) {
```

Comment:
`AgendaItem` contains a lot of information that this `schedule` function doesn’t require. it also makes it less reusable by coupling it to `AgendaItem`s. What do your think about a new data class to encapsulate all the data points that this function would need to schedule a new alarm? Something like this (similar to the parcelable version I already shared 😄 ):

```
data class NotificationData(
    val id: String,
    val title: String,
    val description: String?,
    val notifyAt: LocalDateTime // <— note that the Parcelable version has this as a Long, which is simpler to parse. But LocalDateTime is better to work with in domain layer than a Long for time units
)
```